experimental/sql/src/impl_to_iterators.py

```diff
@@ -50,24 +53,84 @@ class FullTableScanRewriter(RelImplRewriter):
   @op_type_rewrite_pattern
   def match_and_rewrite(self, op: RelImpl.FullTableScanOp,
                         rewriter: PatternRewriter):
-    # TODO: Change this once not only SampleInputOp is supported
+    # TODO: Change this once loading from input is supported
     rewriter.replace_matched_op(
-        it.SampleInputOp.get(self.convert_bag(op.result.typ)))
+        it.ConstantStreamOp.get([[IntegerAttr.from_int_and_width(0, 32)],
+                                 [IntegerAttr.from_int_and_width(1, 32)],
+                                 [IntegerAttr.from_int_and_width(2, 32)],
+                                 [IntegerAttr.from_int_and_width(3, 32)]],
+                                self.convert_bag(op.result.typ)))
 
 
 @dataclass
 class AggregateRewriter(RelImplRewriter):
 
   @op_type_rewrite_pattern
   def match_and_rewrite(self, op: RelImpl.Aggregate, rewriter: PatternRewriter):
-    rewriter.replace_matched_op(it.ReduceOp.get(op.input.op))
+    rewriter.replace_matched_op(
+        it.ReduceOp.get(op.input.op, StringAttr.from_str("sum_struct"),
+                        self.convert_bag(op.result.typ)))
 
 
 #===------------------------------------------------------------------------===#
 # Conversion setup
 #===------------------------------------------------------------------------===#
 
 
+def add_reduce_functions(ctx: MLContext, mod: ModuleOp):
```

Comment:
It looks like you intend to predefine all possible reduce functions. However, I think that that approach doesn't scale: there is an infinite number of possible schemas we'd want to be able to run on and each of them would result in a different function (per SQL function per field). I was imagining your lowering to produce a reduce function specifically for each reduce op that it lowers (with local visibility and with a unique name that may or may not reflect what it does, as we discussed).

If this is what you'll want to do eventually and this is just a shortcut, I suggest you document this (with a TODO).